progressutil/iocopy_test.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2016 CoreOS Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package progressutil
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"testing"
+	"time"
+)
+
+type fakeReader struct {
+	input chan []byte
+	done  bool
+}
+
+func (fr *fakeReader) Read(p []byte) (int, error) {
+	if fr.done {
+		return 0, io.EOF
+	}
+	i := copy(p, <-fr.input)
+	return i, nil
+}
+
+type fakeWriter struct {
+	received *bytes.Buffer
```

Comment:
Actually after looking at this again, the entire `fakeWriter` is pointless. I'll just use a `bytes.Buffer` directly instead.